psydac/fem/projectors.py

```diff
@@ -158,3 +157,258 @@ def get_dual_dofs(Vh, f, domain_h, backend_language="python", return_format='ste
         return tilde_f.toarray()
     else:
         return tilde_f
+
+#===============================================================================
+class DirichletBoundaryProjector(LinearOperator):
+
+    def __init__(self, fem_space, bcs=None, space_kind=None):
+
+        assert isinstance(fem_space, FemSpace)
```

Comment:
How about `bcs` and `space_kind`?

```suggestion
        assert isinstance(fem_space, FemSpace)
        assert bcs is None or isinstance(bcs, Iterable)
        assert space_kind is None or isinstance(space_kind, (str, SpaceType))
```

`Iterable` can be imported from the module `collections.abc`, or `typing`.